### PR TITLE
Adjust tests so they run on 4.10 and 4.11

### LINKED ISSUE #123: Tests fail under GAP master branch
Please see e.g. https://travis-ci.org/gap-infra/gap-docker-pkg-tests-master/jobs/597455020

```
########> Diff in /home/gap/inst/gap-master/pkg/datastructures-0.2.4/tst/hashf\
unctions/recursive.tst:56
# Input is:
HashBasic(SymmetricGroup(3));
# Expected output:
Error, Unable to hash object (component)
# But found:
Error, Unable to hash component object
########
       4 ms (0 ms GC) and 383KB allocated for hashfunctions/recursive.tst
testing: /home/gap/inst/gap-master/pkg/datastructures-0.2.4/tst/hashfunctions/\
string.tst
      99 ms (3 ms GC) and 17.0MB allocated for hashfunctions/string.tst
testing: /home/gap/inst/gap-master/pkg/datastructures-0.2.4/tst/hashfunctions/\
trans.tst
    1043 ms (913 ms GC) and 340MB allocated for hashfunctions/trans.tst
testing: /home/gap/inst/gap-master/pkg/datastructures-0.2.4/tst/hashmap.tst
########> Diff in /home/gap/inst/gap-master/pkg/datastructures-0.2.4/tst/hashm\
ap.tst:238
# Input is:
DS_Hash_Value(infinity, 1);
# Expected output:
Error, <ht> must be a hashmap object (not a object (positional))
# But found:
Error, <ht> must be a hashmap object (not a positional object)
########
```

### PR DESCRIPTION
**Pull request by ChrisJefferson**
Fixes #123 

Change the type we use for testing to function, as we don't really care what it is (as long as it is something we don't handle) and function's printing hasn't changed.

### CODE REVIEW

File: tst/hashmap.tst [hunk 1]
```diff
@@ -235,8 +235,8 @@ Error, <key> must not be equal to 'fail'
 # a positional object which isn't a hash map
 gap> IsPositionalObjectRep(infinity);
 true
-gap> DS_Hash_Value(infinity, 1);
-Error, <ht> must be a hashmap object (not a object (positional))
+gap> DS_Hash_Value(x -> x, 1);
+Error, <ht> must be a hashmap object (not a function)
```
Comment (fingolfin) [inline tst/hashmap.tst]: Actually, that test intentionally used a positional object that's not a hashmap (see also the preceding lines). Of course we can drop it, but then this test and the one before it should be deleted

Comment (ChrisJefferson) [reply tst/hashmap.tst]: You are right, I should have read the context.

File: tst/hashfunctions/recursive.tst [hunk 2]
```diff
@@ -53,5 +53,10 @@ gap> HashBasic(rec1) = HashBasic(rec2);
 true
 
 # Check objects we don't handle
+#@if GAPInfo.KernelInfo.KERNEL_API_VERSION >= 7000
```
Comment (fingolfin) [inline tst/hashfunctions/recursive.tst]: But isn't this feature new in GAP 4.11? I.e., won't help here

Comment (fingolfin) [reply tst/hashfunctions/recursive.tst]: I went through all these ideas when I looked at the test failures after they were reported. Only solutions I could come up with:

1. drop the test,
2. adjust our error function to not use TNAM_OBJ directly but rather our own wrapper func which handles T_POSOBJ (and perhaps more) special (and uniformly across all GAP versions).
3. move the problematic tests to a separate .tst file, and then use the 'exclude' feature of TestDirectory to only run it in GAP >= 4.11

Comment (ChrisJefferson) [reply tst/hashfunctions/recursive.tst]: I'm tempted to drop the test, as I'm not sure it's worth the pain of the other options

**Comment by alex-konovalov** [comment]
Your Travis CI setup does not run them in stable-4.11 - I've added it in #126. It would be nice to fix the test and make a package release before GAP 4.11. Happy with any solution, including dropping the test.

File: tst/hashfunctions/recursive.tst (hunk 2 above)
Comment (fingolfin) [reply tst/hashfunctions/recursive.tst]: I agree and made PR #127 to do just that.

**Pull request closed without merging**